backend/fetcher.py

```diff
@@ -160,193 +159,156 @@ def fetch_spec_file(self) -> Optional[dict[str, str]]:
-
-    def _fetch_spec_file_from_task_id(self) -> Optional[dict[str, str]]:
-        with get_temporary_dir() as temp_dir:
-            srpm_url = self._get_srpm_url_from_task()
-            resp = requests.get(srpm_url)
-            try:
-                resp.raise_for_status()
-            except HTTPError as exc:
-                raise FetchError(
-                    "No spec file found in koji for build/task id "
-                    f"#{self.build_or_task_id} and arch {self.arch}."
-                    f"Reason: {exc}"
-                ) from exc
-
-            destination = Path(f'{temp_dir}/{srpm_url.split("/")[-1]}')
-            with open(destination, "wb") as srpm_f:
-                srpm_f.write(resp.content)
-
-            return self._get_spec_file_content_from_srpm(destination, temp_dir)
-
-    def _fetch_spec_file_from_build_id(self) -> Optional[dict[str, str]]:
-        koji_build = self.client.getBuild(self.build_or_task_id)
-        srpm_url = (
-            f"{self.koji_url}/packages/{koji_build['package_name']}"
-            f"/{koji_build['version']}/{koji_build['release']}/src/{koji_build['nvr']}"
-            ".src.rpm"
-        )
-        response = requests.get(srpm_url)
-        with get_temporary_dir() as temp_dir:
-            koji_srpm_path = temp_dir / f"koji_{self.build_or_task_id}.src.rpm"
-            with open(koji_srpm_path, "wb") as src_rpm:
-                src_rpm.write(response.content)
-                return self._get_spec_file_content_from_srpm(koji_srpm_path, temp_dir)
-
     @handle_errors
     def fetch_spec_file(self) -> dict[str, str]:
-        if self._is_build_id:
-            fetch_spec_fn = self._fetch_spec_file_from_build_id
-        else:
-            fetch_spec_fn = self._fetch_spec_file_from_task_id
+        try:
+            request_url = self.get_task_request_url()
+            package_name = re.findall(r"/rpms/(.+)\.git", request_url)[0]
+            commit_hash = re.findall(r"\.git#(.+)$", request_url)[0]
+            spec_url = "https://src.fedoraproject.org/rpms/" \
+                f"{package_name}/raw/{commit_hash}/f/{package_name}.spec"
+            response = requests.get(spec_url)
+            response.raise_for_status()
+            spec_dict = {"name": f"{package_name}.spec", "content": response.text}
+        except HTTPError as exc:
+            raise FetchError(
+                "No spec file found in koji for task "
+                f"#{self.task_id} and arch {self.arch}."
+                f"Reason: {exc}"
+            ) from exc
 
-        spec_dict = fetch_spec_fn()
         if spec_dict is None:
             raise FetchError(
-                "No spec file found in koji for build/task id "
-                f"#{self.build_or_task_id} and arch {self.arch}"
+                f"No spec file found in koji for build {self.build_id}, task "
+                f"#{self.task_id} and arch {self.arch}"
             )
```

Comment:
I think this can be deleted since spec_dict can't be None anymore

Reply:
I hope so :D will delete it